src/main/java/ee/tuleva/onboarding/comparisons/fundvalue/retrieval/GlobalStockIndexCreator.java

```diff
@@ -0,0 +1,95 @@
+package ee.tuleva.onboarding.comparisons.fundvalue.retrieval;
+
+import ee.tuleva.onboarding.comparisons.fundvalue.FundValue;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Slf4j
+@Service
+public class GlobalStockIndexCreator implements ComparisonIndexRetriever {
+    public static final String KEY = "NEW_GLOBAL_STOCK_INDEX";
+
+    NamedParameterJdbcTemplate jdbcTemplate;
+
+    private static final String SQL = "" +
+        "SELECT * " +
+        "FROM ( " +
+        "(SELECT * FROM index_values WHERE key='MARKET' and date <= '2019-12-31' ORDER BY date DESC LIMIT 4) " +
+        "UNION " +
+        "(SELECT * FROM index_values WHERE key='GLOBAL_STOCK_INDEX' and date >='2020-01-01' LIMIT 4) " +
+        ") values ORDER BY values.date ASC";
+
+    private static final class FundValueRowMapper implements RowMapper<FundValue> {
+        @Override
+        public FundValue mapRow(ResultSet rs, int rowNum) throws SQLException {
+            return new FundValue(rs.getString("key"), rs.getDate("date").toLocalDate(), rs.getBigDecimal("value"));
+        }
+    }
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    @Override
+    public List<FundValue> retrieveValuesForRange(LocalDate startDate, LocalDate endDate) {
+        List<FundValue> stockValues = getStockValues();
+        return stockValues.stream().filter(fundValue -> {
+            LocalDate date = fundValue.getDate();
+            return (startDate.isBefore(date) || startDate.equals(date)) && (endDate.isAfter(date) || endDate.equals(date));
+        }).collect(toList());
+    }
+
+    private List<FundValue> getStockValues() {
+        try {
+            return createStockList();
+        } catch (IOException e) {
+            throw new IllegalStateException("Could not get Global Stock Index values", e);
+        }
+    }
+
+
+    private List<FundValue> createStockList() throws IOException {
```

Comment:
`createStockList` is not a good method name. Generally speaking, you should never use the name of the data structure (i.e. `List` in this case) in method/variable names. `createStocks` would be better since it gives you flexibility to change the underlying data structure without risking accidentally contradicting (or deceiving) your readers. I.e. if you happen to change the return type of the method to a `Map` in the future, but forget to change the name of the method then you are effectively lying about the return type of your method (i.e. a `createList()` method actually creates a `Map`). So to avoid this kind of human errors in the future, it's good practice to never expose the return type in your method names (or data types in variable names).

Also a mandatory read:
Clean Code Chapter 2 - Writing Meaningful Names